### PR TITLE
Bug 1118786: Prep deployment and crons for Python 2.7

### PR DESCRIPTION
**Pull request by pmclanahan**
This changes cron files to templates that get processed into the cronfiles. Only outstanding question is whether removing the original files then regenerating them will permanently break the existing symlinks. I believe it won't, but that's all I could think of that might break.

### CODE REVIEW

**Comment by pmclanahan** [comment]
@jgmize r?

**Comment by pmclanahan** [comment]
Doh! Forgot we had tests for cron files. Working on it.

**Comment by pmclanahan** [comment]
WOO! Fixed.

@jgmize fixed it by having tests generate the cronfiles every test run. Thinking about putting a skip on these unless it's on Travis. Thoughts? They're still quite fast.

File: etc/cron.d/bedrock-prod.tmpl [hunk 1]
```diff
@@ -0,0 +1,28 @@
+#
+# {{ header }}
+#
+
+MAILTO="webops-cron@mozilla.com,cron-bedrock@mozilla.com"
+
+# webdev requests that prod auto-updates the PHP side, for similar to old
```
Comment (jgmize) [inline etc/cron.d/bedrock-prod.tmpl]: hmm, maybe now is a good time to update this comment :)

File: etc/cron.d/bedrock-prod.tmpl [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+#1-59/15 * * * * {{ user }} /data/bedrock/src/update-www.mozilla.org-product-details.sh
+
+# bug 996144 & 1014586
+2 */6 * * * {{ cron }} update_tweets > /dev/null 2>&1
+
+# bug 1072999, 1109344
+*/30 * * * * {{ django }} update_security_advisories --quiet > /dev/null 2>&1
+
+# bug 1087533
+# TODO: redirect stdout and stderr to null after initial deployment and verification
```
Comment (jgmize) [inline etc/cron.d/bedrock-prod.tmpl]: Seems like it would be trivial to fix this TODO

File: bin/gen-crons.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+    (opts, args) = parser.parse_args()
+
+    if not opts.webapp:
+        parser.error('-w must be defined')
+
+    if not opts.template:
+        parser.error('-t must be defined')
+
+    ctx = {'django': 'cd %s && %s manage.py' % (opts.webapp, opts.python)}
```
Comment (jgmize) [inline bin/gen-crons.py]: nit: seems like this would be better named as 'manage' or 'django_manage' that would make it clearer to read the cron templates.

Comment (pmclanahan) [inline bin/gen-crons.py]: Good call. This was in the original.

**Comment by pmclanahan** [comment]
Updated some things from your notes. Thanks!

**Comment by pmclanahan** [comment]
@jgmize I think this is pretty good now. Shall I push it to a branch on the primary repo and attempt a deploy on a demo? The demo deploy wouldn't trigger a cron build unfortunately, but it'd be a good test regardless.

**Comment by jgmize** [comment]
Actually,  I think we're probably good to merge to master and do a full
test on dev, but if you'd like to test on a demo env first don't let me
stop you. :)

**Comment by pmclanahan** [comment]
Tested on demo1, fixed issue. I'll rebase and we'll be ready for merge shortly.

**Comment by jgmize** [comment]
Great! What was the specific issue?

**Comment by pmclanahan** [comment]
I had forgotten to pass in `ctx` in the `management_cmd` calls in `deploy_dev_base.py`.

**Pull request merged**